### ISSUE TITLE
About the model's generalization ability

### ISSUE THREAD

**Issue by HongRealLee** [protossw512/AdaptiveWingLoss#20]
Thank you for your implementation!
My experiments require facial landmarks annotated in some dataset without ground truth annotation. So I intended to use your model as preprocessing.
Firstly, I ran this model as instructed on WFLW dataset, it did perform perfectly.
Then, I ran it on KDEF dataset, which is a lab conditioned, multi-posed facial dataset. While the landmarks were way from accurate positions. 
Why couldn't it generalize from a obviously difficult dataset to a much simpler one?
![step10](https://user-images.githubusercontent.com/44900082/79638822-34535c80-81ba-11ea-94ff-383def10f56b.png)
![step20](https://user-images.githubusercontent.com/44900082/79638825-36b5b680-81ba-11ea-9c6a-a17976bf595b.png)
![step30](https://user-images.githubusercontent.com/44900082/79638827-37e6e380-81ba-11ea-8b29-a694dcb8f8ef.png)
![step40](https://user-images.githubusercontent.com/44900082/79638828-387f7a00-81ba-11ea-91e8-e2cc4759c1ec.png)

![step10](https://user-images.githubusercontent.com/44900082/79638758-c3ac4000-81b9-11ea-9dbb-288c8a0cb9ed.png)
![step20](https://user-images.githubusercontent.com/44900082/79638761-c60e9a00-81b9-11ea-8717-a038c185504a.png)
![step30](https://user-images.githubusercontent.com/44900082/79638765-c870f400-81b9-11ea-9d53-f858f71a2329.png)
![step40](https://user-images.githubusercontent.com/44900082/79638767-c9a22100-81b9-11ea-9245-1862fc27218f.png)

**Comment by HongRealLee** [comment]
OMG！
I observered these misplaced landmarks closely, and find that their shapes basically match subjects's facial parts,  but squeezed on th y demension.
So I tried to enlarge their y cordinations by the ratio of image itself. The modified landmarks matched perfectly! The model's generalizatin is absolutly fine. 
Then the previous problem may rise from my code modification. I'd better look into my code more carefully, debug and avoid manual modification on my landmarks.
Again, thanks for your sharing! 
![step10](https://user-images.githubusercontent.com/44900082/79829598-0ba7ae80-83d6-11ea-8422-0b1911325995.png)
![step20](https://user-images.githubusercontent.com/44900082/79829603-0d717200-83d6-11ea-9390-b321c1d53a79.png)
